documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -179,13 +216,96 @@ Make sure that the allocator and the execution policy use the same SYCL queue:
     return 0;
   }
 
-For ``std::vector`` with a USM allocator we recommend to use ``std::vector::data()`` in
+Iterators to ``std::vector`` are `indirectly device accessible <indirectly-device-accessible>`_ if and only if a 
+``sycl::usm_allocator`` is used. For ``std::vector`` with a USM allocator we recommend to use ``std::vector::data()`` in
 combination with ``std::vector::size()`` as shown in the example above, rather than iterators to
 ``std::vector``. That is because for some implementations of the C++ Standard Library it might not
 be possible for |onedpl_short| to detect that iterators are pointing to USM-allocated data. In that
 case the data will be treated as if it were in host memory, with an extra copy made to a SYCL buffer.
 Retrieving USM pointers from ``std::vector`` as shown guarantees no unintended copying.
 
+.. _use-iterators:
+
+Use |onedpl_short| Iterators
+----------------------------
+
+|onedpl_short| provides a set of `iterators <iterators-details>`_ that can be used to pass data to algorithms, in
+combination with the data storage types described above. To pass data to an algorithm with a device execution policy,
+use iterators that are both `SYCL device-copyable`_ and `indirectly device accessible <indirectly-device-accessible>`_.
+These properties of the |onedpl_short| iterators are described in `this table <iterator-properties-table>`_.
+
+Use Custom Iterators
```

Comment:
I've just found that the [corresponding RFC](https://github.com/uxlfoundation/oneDPL/tree/main/rfcs/proposed/iterators_pass_directly_customization) is at "proposal" stage, not even "experimental".

Should we rush documenting it for 2022.9 the way it is described right now?

Perhaps, it may become an experimental feature, and we will need to change the interface and move it into "Technical Preview" section.
I think that this feature is for very advanced users and it may be sufficient for such users to reference this RFC. In the meantime, we can polish the documentation.

Reply:
The feature is implemented as a fully supported feature of oneDPL -- https://github.com/uxlfoundation/oneDPL/pull/2126
It is also included in the oneDPL specification as a full feature -- https://github.com/uxlfoundation/oneAPI-spec/pull/625

A PR to move the RFC into the supported folder is needed to get it in the correct state.  Given this I do think we need to include documentation for the feature in 2022.9.

Reply:
https://github.com/uxlfoundation/oneDPL/pull/2278